project/Li-Pro/README.rst

```diff
@@ -0,0 +1,77 @@
+========================
+Columnar Structure Array
+========================
+
+--------
+Abstract
+--------
+
+An **array of structure (AoS)** is not cache-friendly nor SIMD friendly. One of its replacement is **structure of array (SoA)**.
+However, AoS is so straightforward for programmers to code in, and easy to maintain. It is also fundamental for OOP.
+
+This project designs a structure helper for programmers to use SoA, but to code as in AoS.
+
+---------
+Objective
+---------
+- C++ base-level API
+  
+  - Fundamental Types support
+  - Custom type support, possibly at declaration phase
+  - Wrapper for STL containers *(optional)*
+  - Automatic type extension by type structure discovery, with parsers *(optional)*
+
+- Python API
+  
+  - A fixed-type list wrapper for PyObject-s
+  - NumPy- & Pandas- friendly *(optional)*
+
+- Ensure memory layout properties hold
+  
+  We should satisfy `key features of columnar format <https://arrow.apache.org/docs/format/Columnar.html#:~:text=The%20columnar%20format%20has,access%20in%20shared%20memory>`_ ,
+  
+  - Data adjacency for sequential access of same member
+  - Constant-time random access
+  - SIMD and vectorization friendly
+  - Relocatable with zero-copy access
+
+  And addtionally,
+
+  - Compile-time address resolution
+    
+    Hopefully with same instructions as AoS, with light object reference handle (not too much than a pointer).
+    
+  - Optimizations for cache locality, e.g. padding *(optional)*
+
```

Comment:
Please describe what how the basic data structure and API look like.  You may use pseudo code or any form.